tests/k8s.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+
+set -e
+
+bold=$(tput bold)
+norm=$(tput sgr0)
+red=$(tput setaf 1)
+green=$(tput setaf 2)
+yellow=$(tput setaf 3)
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+CONFDIR=$(realpath "${DIR}/../k8s")
+
+MINIKUBEPROFILE="SPIRE-SYSTEMS-TEST"
+MINIKUBECMD="minikube -p ${MINIKUBEPROFILE}"
+CHECKINTERVAL=1
+if [ -n "${TRAVIS}" ]; then
+	# Use the default profile inside of Travis
+	MINIKUBECMD="/usr/local/bin/minikube"
+	# Travis is slow. Give our containers more time.
+	CHECKINTERVAL=5
+fi
+TMPDIR=$(mktemp -d)
+SERVERLOGS=${TMPDIR}/spire-server-logs.log
+
+start_minikube() {
+	# Travis will start up minikube (via .travis.yml)
+	if [ -z "${TRAVIS}" ]; then
+		echo "${bold}Starting minikube... ${norm}"
+		${MINIKUBECMD} start
+		eval $(${MINIKUBECMD} docker-env)
+	fi
+}
+
+tear_down_config() {
+	kubectl delete daemonset spire-agent --namespace spire > /dev/null || true
+	kubectl delete configmap spire-agent --namespace spire > /dev/null || true
+	kubectl delete serviceaccount spire-agent --namespace spire > /dev/null || true
+	kubectl delete deployment spire-server --namespace spire > /dev/null || true
+	kubectl delete service spire-server --namespace spire > /dev/null || true
+	kubectl delete configmap spire-server --namespace spire > /dev/null || true
+	kubectl delete secrets spire-server --namespace spire > /dev/null || true
+	kubectl delete serviceaccount spire-server --namespace spire > /dev/null || true
+	kubectl delete namespace spire > /dev/null || true
```

Comment:
BTW, "delete namespace spire" turns out to be the only command required - deleting a namespace will also delete all objects within it, which is quite nice. I didn't discover this until after I wrote up the tutorial draft. :-/